scripts/tagging_scripts/insert_id_and_ref_tags.py

```diff
@@ -0,0 +1,256 @@
+"""
+Utility script to read the list of topics from mongodb and insert mandatory
+id and reference tags for these topics. The script only works for topics
+with the naming convention /campus/building/device/<optional sub device>/point.
+A error will be displayed for any topics that do not conform to this naming
+convention and will be skipped.
+
+**Configure user configuration variables right below imports before running the
+scripts.**
+
+By default the script only set id field and parent references but if there are
+other common tags that apply to ALL campus, building, device, or point, you
+could update the tags variable in get_campus_tags, get_site_tags,
+get_equip_tags, and get_point_tags method respectively
+"""
+
+import random
+import re
+import sqlite3
+from pymongo import MongoClient
+from pymongo.errors import BulkWriteError
+import datetime
+
+## User configuration - start ##
+
+#Connection string to the mongo db from which list of topics should be read
+topics_connection_string = "mongodb://reader:volttronReader@vc-db.pnl.gov" \
```

Comment:
Pleae stop puting username/passwords in the files.  Use a config file or argparse pass as a parameter!

Reply:
sorry. removed